pkg/chaosdaemon/jvm_server_backup.go

```diff
@@ -0,0 +1,171 @@
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package chaosdaemon
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+	pb "github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+const (
+	bytemanHome            = "/usr/local/byteman/"
+	bmInstallCommandBackUp = bytemanHome + "bin/bminstall.sh -b -Dorg.jboss.byteman.transform.all -Dorg.jboss.byteman.verbose -Dorg.jboss.byteman.compile.to.bytecode -p %d %d"
+	bmSubmitCommandBackUp  = bytemanHome + "bin/bmsubmit.sh -p %d -%s %s"
+)
+
+func (s *DaemonServer) InstallJVMRulesBackUp(ctx context.Context,
+	req *pb.InstallJVMRulesRequest) (*empty.Empty, error) {
+	pid, err := s.crClient.GetPidFromContainerID(ctx, req.ContainerId)
+	if err != nil {
+		log.Error(err, "GetPidFromContainerID")
+		return nil, err
+	}
+
+	//todo get pid in the container，
+	processBuilder := bpm.DefaultProcessBuilder("sh", "-c", "ps -ef | grep java | grep -v grep | awk 'NR==1 {print $2}'").SetContext(ctx).SetNS(pid, bpm.MountNS).SetNS(pid, bpm.PidNS)
+	output, err := processBuilder.Build().Output()
+	if err != nil {
+		return nil, err
+	}
+	log.Info("get java pid", "output", string(output))
+
+	javaPid, err := strconv.Atoi(strings.Replace(string(output), "\n", "", -1))
+	if err != nil {
+		return nil, err
+	}
+
+	// todo: Need to write the BYTEMAN_HOME environment variable in bminstall.sh and bmsubmit.sh
+	// or do this in code ?
+	agentFile, err := os.Open("/usr/local/bin/byteman.tar.gz")
+	if err != nil {
+		return nil, err
+	}
+	//processBuilder = bpm.DefaultProcessBuilder("sh", "-c", "cat > /usr/local/byteman/lib/byteman.jar").SetContext(ctx)
+	processBuilder = bpm.DefaultProcessBuilder("sh", "-c", "cat > /usr/local/byteman.tar.gz").SetContext(ctx)
+	processBuilder = processBuilder.SetNS(pid, bpm.MountNS).SetStdin(agentFile)
+	output, err = processBuilder.Build().CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	if len(output) > 0 {
+		log.Info("copy byteman.tar.gz", "output", string(output))
+	}
+	// uncompress byteman.tar.gz
+	processBuilder = bpm.DefaultProcessBuilder("sh", "-c", "tar -zxf /usr/local/byteman.tar.gz -C /usr/local").SetContext(ctx).SetNS(pid, bpm.MountNS)
+	output, err = processBuilder.Build().CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	if len(output) > 0 {
+		log.Info("tar byteman.tar.gz", "output", string(output))
+	}
+
+	bmInstallCmd := fmt.Sprintf(bmInstallCommandBackUp, req.Port, javaPid)
+
+	output, err = s.crClient.ExecCommandByContainerID(ctx, req.ContainerId, []string{"sh", "-c", bmInstallCmd})
```

Comment:
Is it necessary to execute the script through container runtime? The `bpm` package supports setting the namespaces (e.g. `.SetNS(pid, bpm.MountNS)` above).

Reply:
I tried it, but got a problem with environment variables like `JAVA_HOME`。I remember that a similar issue was raised under the [nsexec](https://github.com/chaos-mesh/nsexec) project before. https://github.com/chaos-mesh/nsexec/issues/1

Reply:
I prefer to get the environment variables from the procfs (`/proc/PID/environ`) or container runtime, set the environment variables of the subprocess, then use the `nsexec`.

Reply:
I guess you mean is this 
``` go
	//todo
	processBuilder = bpm.DefaultProcessBuilder("sh", "-c", bmInstallCmd).SetContext(ctx)
	if err = setEnvsToProcess(processBuilder, strconv.Itoa(int(pid))); err != nil {
		return nil, err
	}
	processBuilder = processBuilder.SetNS(pid, bpm.MountNS).SetStdin(agentFile)
	output, err = processBuilder.Build().CombinedOutput()
```
set Envs to Process
```go
func setEnvsToProcess(processBuilder *bpm.ProcessBuilder, pid string) error {
	//todo ,get env by pid from /proc/pid/environ
	bEnvs, err := util.GetEnvsByProcess(strconv.Itoa(int(pid)))
	if err != nil {
		return err
	}
      //split  envs content by []byte{0}
	for _, env := range util.SplitEnvs(bEnvs) {
		envMap := strings.Split(env, "=")
		processBuilder.SetEnv(envMap[0], envMap[1])
	}
	return nil
}
```
right?

Reply:
prompt after I try
> 	{"error": "fork/exec /usr/local/bin/nsexec: invalid argument"}

Reply:
> I guess you mean is this

Yes 🍻 

> {"error": "fork/exec /usr/local/bin/nsexec: invalid argument"}

Have you uploaded the code? Maybe I could debug for you 🤔 . I guess there is something unexpected inside the environment variables.

Reply:
code has been commited,thanks for your help。🌹

Reply:
@zhqqqy I have fixed (and added some tests), and the code has been pushed, please check whether it works 😄 .

Reply:
Unfortunately, rule submit failed, prompt `Connection refused`
![image](https://user-images.githubusercontent.com/13046318/156732493-eb951d3e-63d3-4149-98ac-77acc814028b.png)
the byteman has been attahed to target proocess
![image](https://user-images.githubusercontent.com/13046318/156732904-164ca39c-17e4-4dd6-a3bf-24cf9f996d27.png)

Reply:
Fixed. The `bmsubmit.sh` script should be run inside the target network namespace (see my latest commit).

Reply:
> Fixed. The `bmsubmit.sh` script should be run inside the target network namespace (see my latest commit).

great, works perfectly!